direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BulkBlobWriter.java

```diff
@@ -74,12 +76,12 @@ public void write(StreamElement data, long watermark, CommitCallback statusCallb
 
   @Override
   public void updateWatermark(long watermark) {
-    wrap.updateWatermark(watermark);
+    Optional.ofNullable(wrap).ifPresent(w -> w.updateWatermark(watermark));
```

Comment:
these kind of extra allocations in internal methods are really uncessary, and don't add anything to overall readability (I'd say readability is even worse)

The correct construct should be.
```
if (wrap != null) {
  wrap.updateWatermark(watermark)
}
```

Is there ANY advantage of using your approach?

Reply:
that is only a matter of taste